File: solutions/0217-contains-duplicate/solution.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:

        if len(nums) != len(set(nums)):
            return True
        return False

        # d = {}

        # for num in nums:

        #     if num in d:
        #         return True
        #     else:
        #         d[num] = 0
        
        # return False
        

File: solutions/0217-contains-duplicate/solution.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer, Integer> numToCount = new HashMap<Integer, Integer>();

        int i = 0;
        while (i < nums.length) {
            if (!numToCount.containsKey(nums[i])) {
                numToCount.put(nums[i], 0);
            }
            numToCount.put(nums[i], numToCount.get(nums[i]) + 1);
            if (numToCount.get(nums[i]) > 1) {
                return true;
            }
            i++;
        }
        return false;
    }
}
